server/models/contests/getContest.js

```diff
@@ -0,0 +1,33 @@
+const { pool } = require('../../models/database')
+
+/**
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {Object} param0.params
+ * @return {Promise}
+ */
+
+function getContest({ username, params }) {
+  return new Promise((resolve, reject) => {
+    const { contest_id: contestId } = params
+    pool.query(
+      `SELECT * FROM contests WHERE id=? 
+      AND (public=1 OR EXISTS(SELECT 1 FROM contests_groups cg INNER JOIN user_groups ug ON cg.group_id = ug.group_id 
```

Comment:
If a moderator requests to this end point, it would not return anything.
So, also allow contest moderators to `getContest()`

Reply:
If that seems complex, we can do one thing:
```
Create a separate end-point for moderators to view details of a contest, like
/contests/:contest_id/details
```

Reply:
And this endpoint should only return the following information then: (instead of SELECT *, only SELECT these)
  id
  creator
  name
  start_time
  end_time
  about
  rules
  prizes
   participants_count

Reply:
And the end-point which would be for moderators, would return all details about that contest, including the group name and group id of all the groups eligible for that contest